pygmt/src/subplot.py

```diff
@@ -0,0 +1,215 @@
+        any side if these should be typeset as part of the tag. Use **+j|J**\
+        *refpoint* to specify where the tag should be placed in the subplot
+        [TL]. Note: **+j** sets the justification of the tag to *refpoint*
+        (suitable for interior tags) while **+J** instead selects the mirror
+        opposite (suitable for exterior tags). Append **+c**\ *dx*[/*dy*] to
+        set the clearance between the tag and a surrounding text box requested
+        via **+g** or **+p** [3p/3p, i.e., 15% of the :gmt-term:`FONT_TAG` size
+        dimension]. Append **+g**\ *fill* to paint the tag's text box with
+        *fill* [no painting]. Append **+o**\ *dx*\ [/*dy*] to offset the tag's
+        reference point in the direction implied by the justification [4p/4p,
+        i.e., 20% of the :gmt-term:`FONT_TAG` size]. Append **+p**\ *pen* to
+        draw the outline of the tag's text box using selected *pen* [no
+        outline]. Append **+r** to typeset your tag numbers using lowercase
+        Roman numerals; use **+R** for uppercase Roman numerals [Arabic
+        numerals]. Append **+v** to increase tag numbers vertically down
+        columns [horizontally across rows].
+    {B}
+    clearance : str
+        [*side*]\ *clearance*.
+        Reserve a space of dimension *clearance* between the margin and the
+        subplot on the specified side, using *side* values from **w**, **e**,
+        **s**, or **n**, or **x** for both **w** and **e** or **y** for both
+        **s** and **n**.  No *side* means all sides. The option is repeatable
+        to set aside space on more than one side. Such space will be left
+        untouched by the main map plotting but can be accessed by modules that
+        plot scales, bars, text, etc.  Settings specified under **begin**
+        directive apply to all subplots, while settings under **set** only
+        apply to the selected (active) subplot. **Note**: Common options
+        **x_offset** and **y_offset* are not available during subplots; use
+        **clearance** instead.
+    {J}
+    margins : list
+        This is margin space that is added between neighboring subplots (i.e.,
+        the interior margins) in addition to the automatic space added for tick
+        marks, annotations, and labels. The margins can be specified as either:
+
+        - a single value (for same margin on all sides). E.g. '5c'.
+        - a pair of values (for setting separate horizontal and vertical
+          margins). E.g. ['5c', '3c'].
+        - a set of four values (for setting separate left, right, bottom, and
+          top margins). E.g. ['1c', '2c', '3c', '4c'].
+
+        The actual gap created is always a sum of the margins for the two
+        opposing sides (e.g., east plus west or south plus north margins)
+        [Default is half the primary annotation font size, giving the full
+        annotation font size as the default gap].
+    {R}
+    layout : str or list
```

Comment:
Can we do `sharex` and `sharey`, similar to matplotlib subplot?

Reply:
Good idea, I'll look into this, would be good to split up the layout (S) docs actually into x and y components.